06-Job-Declaration-Protocol.md

```diff
@@ -103,9 +102,49 @@ The table below shows a comparison between the two Sv2 Job Declation Modes:
 | knowledge of txdata on the template | • JDC | • JDC <br> • JDS (on behalf of Pool) |
 | ability to broadcast mined block    | • JDC | • JDC <br> • JDS (on behalf of Pool) |
 
-## 6.4 Job Declaration Protocol Messages
+## 6.4 Custom Work Rejection
 
-### 6.4.1 `SetupConnection` Flags for Job Declaration Protocol
+This section aims to clarify the different ways in which work can be rejected under Job Declaration.
+
+## 6.4.1 Template Rejection
+
+This can also be generally referred as the rejection of a custom job, or rejection of custom work.
+
+Under Full-Template mode, JDS can reply with a `DeclareMiningJob.Error` to signal that the Pool is not willing to reward work based on the declared template. That could happen for different reasons, to name a few:
```

Comment:
Isn't it important to mention again here that miner would never stop mining their prefered template and would fallback? I feel like here we need to be careful how we frame things as it can easily get taken out of context by keyboard warriors on X.

Reply:
What happens in the Coinbase-only mode?